In [81]:
import pandas as pd
import numpy as np

In [82]:
df1 = pd.read_csv("/content/phoneProduct.csv")

In [83]:
df1

title           brand  \
0          Điện thoại Samsung Galaxy S25 5G 12GB/256GB         Samsung   
1    Điện thoại Samsung Galaxy S25 Ultra 5G 12GB/256GB         Samsung   
2                   Điện thoại iPhone 16 Pro Max 256GB  iPhone (Apple)   
3                       Điện thoại iPhone 16 Pro 128GB  iPhone (Apple)   
4    Điện thoại Samsung Galaxy S24 Ultra 5G 12GB/256GB         Samsung   
..                                                 ...             ...   
105                     Điện thoại iPhone 15 Pro 512GB  iPhone (Apple)   
106                      Điện thoại OPPO A58 8GB/128GB            OPPO   
107                           Điện thoại Mobell Rock 4          Mobell   
108              Điện thoại OPPO Find N3 5G 16GB/512GB            OPPO   
109   Điện thoại Samsung Galaxy S25 Plus 5G 12GB/256GB         Samsung   

           price                                          link_item  \
0    19.990.000₫  https://www.thegioididong.com/dtdd/samsung-gal...   
1    30.990.000₫  https://www.thegioididong.com/dtdd/samsung-gal...   
2    32.890.000₫  https://www.thegioididong.com/dtdd/iphone-16-p...   
3    27.590.000₫   https://www.thegioididong.com/dtdd/iphone-16-pro   
4    25.990.000₫  https://www.thegioididong.com/dtdd/samsung-gal...   
..           ...                                                ...   
105  32.490.000₫  https://www.thegioididong.com/dtdd/iphone-15-p...   
106   4.490.000₫  https://www.thegioididong.com/dtdd/oppo-a58-4g...   
107     790.000₫   https://www.thegioididong.com/dtdd/mobell-rock-4   
108  36.990.000₫  https://www.thegioididong.com/dtdd/oppo-find-n...   
109  23.990.000₫  https://www.thegioididong.com/dtdd/samsung-gal...   

                                              link_img        resolution  \
0    https://cdn.tgdd.vn/Products/Images/42/333363/...          Full HD+   
1    https://cdn.tgdd.vn/Products/Images/42/333347/...    Quad HD+ (2K+)   
2    https://cdn.tgdd.vn/Products/Images/42/329149/...  Super Retina XDR   
3    https://cdn.tgdd.vn/Products/Images/42/329143/...  Super Retina XDR   
4    https://cdn.tgdd.vn/Products/Images/42/307174/...    Quad HD+ (2K+)   
..                                                 ...               ...   
105  https://cdn.tgdd.vn/Products/Images/42/303832/...  Super Retina XDR   
106  https://cdn.tgdd.vn/Products/Images/42/311354/...          Full HD+   
107  https://cdn.tgdd.vn/Products/Images/42/288631/...              QVGA   
108  https://cdn.tgdd.vn/Products/Images/42/302953/...    Quad HD+ (2K+)   
109  https://cdn.tgdd.vn/Products/Images/42/333359/...    Quad HD+ (2K+)   

                        size     rate          Lượt bán  index_  \
0                       6.2"  No rate  No quantity sale       1   
1                       6.9"  No rate  No quantity sale       2   
2                       6.9"      4.9      Đã bán 94,5k       3   
3                       6.3"      4.9      Đã bán 10,4k       4   
4                       6.8"      4.9      Đã bán 24,2k       5   
..                       ...      ...               ...     ...   
105                     6.1"      4.9        Đã bán 729     106   
106                    6.72"      4.9     Đã bán 123,4k     107   
107                     2.4"      4.9      Đã bán 11,3k     108   
108  Chính 7.82" & Phụ 6.31"        5        Đã bán 372     109   
109                     6.7"  No rate  No quantity sale     110   

     discount_index original price discount percent  
0               1.0    22.990.000₫             -13%  
1               2.0    33.990.000₫              -8%  
2               3.0    34.990.000₫              -6%  
3               4.0    28.990.000₫              -4%  
4               5.0    29.990.000₫             -13%  
..              ...            ...              ...  
105             NaN            NaN              NaN  
106             NaN            NaN              NaN  
107             NaN            NaN              NaN  
108             NaN            NaN              N

In [84]:
df1.dtypes

title                object
brand                object
price                object
link_item            object
link_img             object
resolution           object
size                 object
rate                 object
Lượt bán             object
index_                int64
discount_index      float64
original price       object
discount percent     object
dtype: object

In [85]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   title             110 non-null    object 
 1   brand             110 non-null    object 
 2   price             110 non-null    object 
 3   link_item         110 non-null    object 
 4   link_img          110 non-null    object 
 5   resolution        110 non-null    object 
 6   size              110 non-null    object 
 7   rate              110 non-null    object 
 8   Lượt bán          110 non-null    object 
 9   index_            110 non-null    int64  
 10  discount_index    37 non-null     float64
 11  original price    37 non-null     object 
 12  discount percent  37 non-null     object 
dtypes: float64(1), int64(1), object(11)
memory usage: 11.3+ KB


Xóa đi 2 cột không cần thiết: index_, discount_index
<br> Để thuận tiện hơn trong việc đánh xử lí và đánh giá:
<br>Đổi tên các cột: '**tittle**' thành '**name**', '**Lượt bán**' thành '**quantity_sale**', '**price**' thành '**price (đ)**', '**original price**' thành '**original price (đ)**'


In [86]:
df1.drop(columns=['index_', 'discount_index'], inplace=True)
df1.rename(columns={'title': 'name', 'Lượt bán': 'quantity_sale', 'price': 'price (đ)', 'original price': 'original price (đ)'}, inplace=True)
df1

name           brand  \
0          Điện thoại Samsung Galaxy S25 5G 12GB/256GB         Samsung   
1    Điện thoại Samsung Galaxy S25 Ultra 5G 12GB/256GB         Samsung   
2                   Điện thoại iPhone 16 Pro Max 256GB  iPhone (Apple)   
3                       Điện thoại iPhone 16 Pro 128GB  iPhone (Apple)   
4    Điện thoại Samsung Galaxy S24 Ultra 5G 12GB/256GB         Samsung   
..                                                 ...             ...   
105                     Điện thoại iPhone 15 Pro 512GB  iPhone (Apple)   
106                      Điện thoại OPPO A58 8GB/128GB            OPPO   
107                           Điện thoại Mobell Rock 4          Mobell   
108              Điện thoại OPPO Find N3 5G 16GB/512GB            OPPO   
109   Điện thoại Samsung Galaxy S25 Plus 5G 12GB/256GB         Samsung   

       price (đ)                                          link_item  \
0    19.990.000₫  https://www.thegioididong.com/dtdd/samsung-gal...   
1    30.990.000₫  https://www.thegioididong.com/dtdd/samsung-gal...   
2    32.890.000₫  https://www.thegioididong.com/dtdd/iphone-16-p...   
3    27.590.000₫   https://www.thegioididong.com/dtdd/iphone-16-pro   
4    25.990.000₫  https://www.thegioididong.com/dtdd/samsung-gal...   
..           ...                                                ...   
105  32.490.000₫  https://www.thegioididong.com/dtdd/iphone-15-p...   
106   4.490.000₫  https://www.thegioididong.com/dtdd/oppo-a58-4g...   
107     790.000₫   https://www.thegioididong.com/dtdd/mobell-rock-4   
108  36.990.000₫  https://www.thegioididong.com/dtdd/oppo-find-n...   
109  23.990.000₫  https://www.thegioididong.com/dtdd/samsung-gal...   

                                              link_img        resolution  \
0    https://cdn.tgdd.vn/Products/Images/42/333363/...          Full HD+   
1    https://cdn.tgdd.vn/Products/Images/42/333347/...    Quad HD+ (2K+)   
2    https://cdn.tgdd.vn/Products/Images/42/329149/...  Super Retina XDR   
3    https://cdn.tgdd.vn/Products/Images/42/329143/...  Super Retina XDR   
4    https://cdn.tgdd.vn/Products/Images/42/307174/...    Quad HD+ (2K+)   
..                                                 ...               ...   
105  https://cdn.tgdd.vn/Products/Images/42/303832/...  Super Retina XDR   
106  https://cdn.tgdd.vn/Products/Images/42/311354/...          Full HD+   
107  https://cdn.tgdd.vn/Products/Images/42/288631/...              QVGA   
108  https://cdn.tgdd.vn/Products/Images/42/302953/...    Quad HD+ (2K+)   
109  https://cdn.tgdd.vn/Products/Images/42/333359/...    Quad HD+ (2K+)   

                        size     rate     quantity_sale original price (đ)  \
0                       6.2"  No rate  No quantity sale        22.990.000₫   
1                       6.9"  No rate  No quantity sale        33.990.000₫   
2                       6.9"      4.9      Đã bán 94,5k        34.990.000₫   
3                       6.3"      4.9      Đã bán 10,4k        28.990.000₫   
4                       6.8"      4.9      Đã bán 24,2k        29.990.000₫   
..                       ...      ...               ...                ...   
105                     6.1"      4.9        Đã bán 729                NaN   
106                    6.72"      4.9     Đã bán 123,4k                NaN   
107                     2.4"      4.9      Đã bán 11,3k                NaN   
108  Chính 7.82" & Phụ 6.31"        5        Đã bán 372                NaN   
109                     6.7"  No rate  No quantity sale        12.990.000₫   

    discount percent  
0               -13%  
1                -8%  
2                -6%  
3                -4%  
4               -13%  
..               ...  
105              NaN  
106              NaN  
107              NaN  
108              NaN  
109             -11%  

[110 rows x 11 columns]

Kiểm tra xem có hàng nào bị trùng lặp không

In [87]:
df1.duplicated().sum()

np.int64(0)

Xử lý cột 'size' chuyển sang kiểu số
<br>Vì có một số điện thoại gập hiện này có 2 màn hình nên cột size sẽ có vài dòng có dạng như "Chính 7.82" & Phụ 6.31", trường hợp này sẽ lấy số liệu của màn hình chính
<br>Nếu có giá trị không chuyển được sẽ là NaN

In [88]:
import re

def extract_first_number(text):
    if pd.isna(text):
        return None
    match = re.search(r'(\d+(\.\d+)?)', text)  # Tìm nhóm số (có thể có phần thập phân)
    if match:
        return match.group(1)  # Lấy chuỗi số đầu tiên tìm thấy
    return None

# Áp dụng hàm cho cột 'Size'
df1['size'] = df1['size'].apply(extract_first_number)

df1['size'] = pd.to_numeric(df1['size'], errors='coerce')


In [62]:
df1

name           brand  \
0          Điện thoại Samsung Galaxy S25 5G 12GB/256GB         Samsung   
1    Điện thoại Samsung Galaxy S25 Ultra 5G 12GB/256GB         Samsung   
2                   Điện thoại iPhone 16 Pro Max 256GB  iPhone (Apple)   
3                       Điện thoại iPhone 16 Pro 128GB  iPhone (Apple)   
4    Điện thoại Samsung Galaxy S24 Ultra 5G 12GB/256GB         Samsung   
..                                                 ...             ...   
105                     Điện thoại iPhone 15 Pro 512GB  iPhone (Apple)   
106                      Điện thoại OPPO A58 8GB/128GB            OPPO   
107                           Điện thoại Mobell Rock 4          Mobell   
108              Điện thoại OPPO Find N3 5G 16GB/512GB            OPPO   
109   Điện thoại Samsung Galaxy S25 Plus 5G 12GB/256GB         Samsung   

       price (đ)                                          link_item  \
0    19.990.000₫  https://www.thegioididong.com/dtdd/samsung-gal...   
1    30.990.000₫  https://www.thegioididong.com/dtdd/samsung-gal...   
2    32.890.000₫  https://www.thegioididong.com/dtdd/iphone-16-p...   
3    27.590.000₫   https://www.thegioididong.com/dtdd/iphone-16-pro   
4    25.990.000₫  https://www.thegioididong.com/dtdd/samsung-gal...   
..           ...                                                ...   
105  32.490.000₫  https://www.thegioididong.com/dtdd/iphone-15-p...   
106   4.490.000₫  https://www.thegioididong.com/dtdd/oppo-a58-4g...   
107     790.000₫   https://www.thegioididong.com/dtdd/mobell-rock-4   
108  36.990.000₫  https://www.thegioididong.com/dtdd/oppo-find-n...   
109  23.990.000₫  https://www.thegioididong.com/dtdd/samsung-gal...   

                                              link_img        resolution  \
0    https://cdn.tgdd.vn/Products/Images/42/333363/...          Full HD+   
1    https://cdn.tgdd.vn/Products/Images/42/333347/...    Quad HD+ (2K+)   
2    https://cdn.tgdd.vn/Products/Images/42/329149/...  Super Retina XDR   
3    https://cdn.tgdd.vn/Products/Images/42/329143/...  Super Retina XDR   
4    https://cdn.tgdd.vn/Products/Images/42/307174/...    Quad HD+ (2K+)   
..                                                 ...               ...   
105  https://cdn.tgdd.vn/Products/Images/42/303832/...  Super Retina XDR   
106  https://cdn.tgdd.vn/Products/Images/42/311354/...          Full HD+   
107  https://cdn.tgdd.vn/Products/Images/42/288631/...              QVGA   
108  https://cdn.tgdd.vn/Products/Images/42/302953/...    Quad HD+ (2K+)   
109  https://cdn.tgdd.vn/Products/Images/42/333359/...    Quad HD+ (2K+)   

     size     rate     quantity_sale original price (đ) discount percent  
0    6.20  No rate  No quantity sale        22.990.000₫             -13%  
1    6.90  No rate  No quantity sale        33.990.000₫              -8%  
2    6.90      4.9      Đã bán 94,5k        34.990.000₫              -6%  
3    6.30      4.9      Đã bán 10,4k        28.990.000₫              -4%  
4    6.80      4.9      Đã bán 24,2k        29.990.000₫             -13%  
..    ...      ...               ...                ...              ...  
105  6.10      4.9        Đã bán 729                NaN              NaN  
106  6.72      4.9     Đã bán 123,4k                NaN              NaN  
107  2.40      4.9      Đã bán 11,3k                NaN              NaN  
108  7.82        5        Đã bán 372                NaN              NaN  
109  6.70  No rate  No quantity sale        12.990.000₫             -11%  

[110 rows x 11 columns]

In [89]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   name                110 non-null    object 
 1   brand               110 non-null    object 
 2   price (đ)           110 non-null    object 
 3   link_item           110 non-null    object 
 4   link_img            110 non-null    object 
 5   resolution          110 non-null    object 
 6   size                108 non-null    float64
 7   rate                110 non-null    object 
 8   quantity_sale       110 non-null    object 
 9   original price (đ)  37 non-null     object 
 10  discount percent    37 non-null     object 
dtypes: float64(1), object(10)
memory usage: 9.6+ KB


Xử lý cột 'rate' là điểm đánh giá trung bình của sản phẩm.
Thay thế các gái trị "No rate" bằng giá trị "0". Sau đó chuyển dang kiểu dữ liệu float

In [90]:
df1['rate']=df1['rate'].replace({'No rate': '0'}, regex=True)
df1['rate']=df1['rate'].astype(float)

Xử lý cột 'quantity_sale' số lượt bán của sản phẩm ban đầu có kiểu objact dạng "Đã bán xk" hoặc "Đã bán x" hoặc "No quantity sale" đối với những sản phẩm chưa có lượt bán. Đưa về kiểu số

In [91]:
import re

def clean_quantity(q):
    if "No quantity sale" in q:  # Trường hợp chưa có lượt bán
        return 0
    match = re.search(r"([\d,.]+)k?", q)  # Tìm số có thể chứa dấu '.' hoặc ',' có thể có ký tự "k"
    if match:
        num = match.group(1).replace(",", ".")  # Đổi dấu phẩy thành dấu chấm cho số thập phân
        return float(num) * 1000 if "k" in q else float(num)  # Chuyển "k" thành 1000
    return 0  # Nếu không có số, trả về 0

# Áp dụng hàm cho cột quantity_sale
df1["quantity_sale"] = df1["quantity_sale"].apply(clean_quantity)

Xử lý cột "price" và "original price".
Xóa đi kí tự 'đ' và đưa về dạng số

In [92]:
df1.isna().sum()

name                   0
brand                  0
price (đ)              0
link_item              0
link_img               0
resolution             0
size                   2
rate                   0
quantity_sale          0
original price (đ)    73
discount percent      73
dtype: int64

In [93]:
df1[df1['size'].isna()]

name brand price (đ)  \
90  Điện thoại Itel it9310  Itel  650.000₫   
97  Điện thoại Itel it9211  Itel  490.000₫   

                                         link_item  \
90  https://www.thegioididong.com/dtdd/itel-it9310   
97  https://www.thegioididong.com/dtdd/itel-it9211   

                                             link_img resolution  size  rate  \
90  https://cdn.tgdd.vn/Products/Images/42/329840/...         3"   NaN   4.9   
97  https://cdn.tgdd.vn/Products/Images/42/329839/...       2.4"   NaN   4.9   

    quantity_sale original price (đ) discount percent  
90         5300.0                NaN              NaN  
97         7000.0                NaN              NaN

Trong quá trình crawl data thì đã có sự cố, đối với 2 chiếc điện thoại cục gạch thì giá trị size lại được chuyển qua cho resolution trong khi Size lại mang giá trị là NaN.
<br>Tiến hành đổi lại cho đúng

In [94]:
# Bước 1: Tạo một mask (điều kiện lọc) cho các dòng size NaN nhưng resolution lại có dạng kích thước
mask = df1['size'].isna() & df1['resolution'].str.contains(r'^\d+(\.\d+)?', na=False)

# Bước 2: Chuyển giá trị từ resolution sang size
df1.loc[mask, 'size'] = df1.loc[mask, 'resolution']

# Bước 3: Đặt lại resolution về NaN (hoặc chuỗi trống) cho những dòng này
df1.loc[mask, 'resolution'] = "No resolution"  # hoặc ""

# Kiểm tra lại kết quả
print(df1[mask][['size', 'resolution']])


    size     resolution
90    3"  No resolution
97  2.4"  No resolution


<ipython-input-94-7f96ae65d04e>:2: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  mask = df1['size'].isna() & df1['resolution'].str.contains(r'^\d+(\.\d+)?', na=False)
<ipython-input-94-7f96ae65d04e>:5: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '['3"' '2.4"']' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.
  df1.loc[mask, 'size'] = df1.loc[mask, 'resolution']


In [95]:
df1.isna().sum()

name                   0
brand                  0
price (đ)              0
link_item              0
link_img               0
resolution             0
size                   0
rate                   0
quantity_sale          0
original price (đ)    73
discount percent      73
dtype: int64

In [96]:
df1.loc[df1['original price (đ)'].isna(),'original price (đ)']="No price"
df1.loc[df1['discount percent'].isna(),'discount percent']="No discount info"

In [97]:
df1.isna().sum()

name                  0
brand                 0
price (đ)             0
link_item             0
link_img              0
resolution            0
size                  0
rate                  0
quantity_sale         0
original price (đ)    0
discount percent      0
dtype: int64

In [98]:
import re

def clean_price(p):
    if "No price" in p or "No discount info" in p:  # Trường hợp không có giá
        return 0
    p = re.sub(r"[^\d]", "", p)  # Xóa mọi ký tự không phải số
    return int(p)  # Chuyển thành số nguyên

df1["price (đ)"] = df1["price (đ)"].apply(clean_price)

df1['original price (đ)']=df1['original price (đ)'].apply(clean_price)

Xử lý cột "discount percent". Bỏ đi các kí tự khác chỉ giữ lại số và đưa về kiểu số

In [99]:
df1['discount percent']=df1['discount percent'].replace({'No discount info': '0'}, regex=True)
df1['discount percent']=df1['discount percent'].str.extract(r"(\d+)")
df1['discount percent']=df1['discount percent'].astype(int)

In [100]:
df1

name           brand  \
0          Điện thoại Samsung Galaxy S25 5G 12GB/256GB         Samsung   
1    Điện thoại Samsung Galaxy S25 Ultra 5G 12GB/256GB         Samsung   
2                   Điện thoại iPhone 16 Pro Max 256GB  iPhone (Apple)   
3                       Điện thoại iPhone 16 Pro 128GB  iPhone (Apple)   
4    Điện thoại Samsung Galaxy S24 Ultra 5G 12GB/256GB         Samsung   
..                                                 ...             ...   
105                     Điện thoại iPhone 15 Pro 512GB  iPhone (Apple)   
106                      Điện thoại OPPO A58 8GB/128GB            OPPO   
107                           Điện thoại Mobell Rock 4          Mobell   
108              Điện thoại OPPO Find N3 5G 16GB/512GB            OPPO   
109   Điện thoại Samsung Galaxy S25 Plus 5G 12GB/256GB         Samsung   

     price (đ)                                          link_item  \
0     19990000  https://www.thegioididong.com/dtdd/samsung-gal...   
1     30990000  https://www.thegioididong.com/dtdd/samsung-gal...   
2     32890000  https://www.thegioididong.com/dtdd/iphone-16-p...   
3     27590000   https://www.thegioididong.com/dtdd/iphone-16-pro   
4     25990000  https://www.thegioididong.com/dtdd/samsung-gal...   
..         ...                                                ...   
105   32490000  https://www.thegioididong.com/dtdd/iphone-15-p...   
106    4490000  https://www.thegioididong.com/dtdd/oppo-a58-4g...   
107     790000   https://www.thegioididong.com/dtdd/mobell-rock-4   
108   36990000  https://www.thegioididong.com/dtdd/oppo-find-n...   
109   23990000  https://www.thegioididong.com/dtdd/samsung-gal...   

                                              link_img        resolution  \
0    https://cdn.tgdd.vn/Products/Images/42/333363/...          Full HD+   
1    https://cdn.tgdd.vn/Products/Images/42/333347/...    Quad HD+ (2K+)   
2    https://cdn.tgdd.vn/Products/Images/42/329149/...  Super Retina XDR   
3    https://cdn.tgdd.vn/Products/Images/42/329143/...  Super Retina XDR   
4    https://cdn.tgdd.vn/Products/Images/42/307174/...    Quad HD+ (2K+)   
..                                                 ...               ...   
105  https://cdn.tgdd.vn/Products/Images/42/303832/...  Super Retina XDR   
106  https://cdn.tgdd.vn/Products/Images/42/311354/...          Full HD+   
107  https://cdn.tgdd.vn/Products/Images/42/288631/...              QVGA   
108  https://cdn.tgdd.vn/Products/Images/42/302953/...    Quad HD+ (2K+)   
109  https://cdn.tgdd.vn/Products/Images/42/333359/...    Quad HD+ (2K+)   

     size  rate  quantity_sale  original price (đ)  discount percent  
0     6.2   0.0            0.0            22990000                13  
1     6.9   0.0            0.0            33990000                 8  
2     6.9   4.9        94500.0            34990000                 6  
3     6.3   4.9        10400.0            28990000                 4  
4     6.8   4.9        24200.0            29990000                13  
..    ...   ...            ...                 ...               ...  
105   6.1   4.9          729.0                   0                 0  
106  6.72   4.9       123400.0                   0                 0  
107   2.4   4.9        11300.0                   0                 0  
108  7.82   5.0          372.0                   0                 0  
109   6.7   0.0            0.0            12990000                11  

[110 rows x 11 columns]

In [101]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 110 entries, 0 to 109
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   name                110 non-null    object 
 1   brand               110 non-null    object 
 2   price (đ)           110 non-null    int64  
 3   link_item           110 non-null    object 
 4   link_img            110 non-null    object 
 5   resolution          110 non-null    object 
 6   size                110 non-null    object 
 7   rate                110 non-null    float64
 8   quantity_sale       110 non-null    float64
 9   original price (đ)  110 non-null    int64  
 10  discount percent    110 non-null    int64  
dtypes: float64(2), int64(3), object(6)
memory usage: 9.6+ KB


Đưa ra các thống kê cơ bản

In [102]:
df1['size'] = pd.to_numeric(df1['size'], errors='coerce')

In [103]:
df1.dtypes

name                   object
brand                  object
price (đ)               int64
link_item              object
link_img               object
resolution             object
size                  float64
rate                  float64
quantity_sale         float64
original price (đ)      int64
discount percent        int64
dtype: object

In [105]:
df1.describe().round(3).transpose()

count         mean          std        min          25%  \
price (đ)           110.0  9797272.727  9582385.207  380000.00  2815000.000   
size                108.0        5.914        1.705       1.77        6.275   
rate                110.0        4.779        0.805       0.00        4.900   
quantity_sale       110.0    36824.509    53838.594       0.00     3475.000   
original price (đ)  110.0  5485727.273  9884347.413       0.00        0.000   
discount percent    110.0        4.327        7.546       0.00        0.000   

                           50%         75%          max  
price (đ)           6990000.00  13590000.0  39990000.00  
size                      6.67         6.7         7.92  
rate                      4.90         4.9         5.00  
quantity_sale         14650.00     44250.0    259700.00  
original price (đ)        0.00   7465000.0  43990000.00  
discount percent          0.00         6.0        33.00

In [106]:
df1.to_csv("cleanedPhoneData.csv", index=False, encoding='utf-8-sig')